Sources/Queryable/Handlers/Queryable+Conditional.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  IfQueryable.swift
+//
+//
+//  Created by Kai Quan Tay on 12/1/24.
+//
+
+import SwiftUI
+
+public struct IfQueryable<Item, Result, QueryContent: View>: View {
+    @ObservedObject private var queryable: Queryable<Item, Result>
+    private var animation: Animation? = nil
+    private var alignment: Alignment = .center
+    private var queryContent: (_ item: Item, _ query: QueryResolver<Result>) -> QueryContent
+
+    public init(
+        _ queryable: Queryable<Item, Result>,
+        animation: Animation? = nil,
+        alignment: Alignment = .center,
+        @ViewBuilder queryContent: @escaping (_ item: Item, _ query: QueryResolver<Result>) -> QueryContent
+    ) {
+        self.queryable = queryable
+        self.animation = animation
+        self.alignment = alignment
+        self.queryContent = queryContent
+    }
```

Comment:
Here, we need an additional initializer for `Queryable<Void, Result>`, just for convenience:

```swift
public init(
    _ queryable: Queryable<Void, Result>,
    @ViewBuilder queryContent: @escaping (_ query: QueryResolver<Result>) -> QueryContent
) where Item == Void {
    self.queryable = queryable
    self.queryContent = { _, query in queryContent(query) }
}
```